packages/framework/src/state/application.ts

```diff
@@ -30,8 +32,44 @@ export const appAtom = atom<ApplicationContextDefinition>(
   defaultApplicationContext,
 );
 
-export const themeAtom = selectAtom(appAtom, (appContext) => {
-  return appContext.application?.theme;
-});
+export const themeAtom = atom<EnsembleThemeModel | undefined>(undefined);
+
+const backingStorage = createJSONStorage<string>(() => sessionStorage);
+
+export const currentThemeAtom = atomWithStorage<string>(
+  "ensemble.currentTheme",
+  "default",
+  backingStorage,
+);
 
 export const envAtom = focusAtom(appAtom, (optic) => optic.prop("env"));
+
+export const useThemeContext = (): {
```

Comment:
This is super confusing don't call this context you have another context in the runtime.